From adfb19c24e39a006048000672c61b573a8559d5a Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Tue, 14 Mar 2023 11:20:07 +0530
Subject: [PATCH] ath12k: Modify request_firmware_nowarn to
 request_firmware_direct

The request_firmware_nowarn searches for the requested file in the
/lib/firmware path and performs additional fall back check to sysfs
to check for the firmware. This takes additional bootup time with
IPQ5332 yocto since the calibration file is not present.
Use request_firmware_direct to reduce the delay in boot.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/core.c b/drivers/net/wireless/ath/ath12k/core.c
index 9dc848f..a61ebe7 100644
--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -442,7 +442,7 @@ const struct firmware *ath12k_core_firmware_request(struct ath12k_base *ab,
 
 	ath12k_core_create_firmware_path(ab, file, path, sizeof(path));
 
-	ret = firmware_request_nowarn(&fw, path, ab->dev);
+	ret = request_firmware_direct(&fw, path, ab->dev);
 	if (ret)
 		return ERR_PTR(ret);
 
-- 
2.17.1

